### ISSUE TITLE
Spike: Discover Root cause of GitHub actions stoppage in January (Before the end of February!)

### ISSUE THREAD

**Issue by mreekie** [IQSS/dataverse#9355]
The best guess so far on this that it's related to usage limits.
We don't know why however.

Quick summary:
We had a Github Actions outage starting on January 27th.
It was reported by @pdurbin & @kuriwaki in [GitHub Actions failing due to spending limits #125](https://github.com/IQSS/dataverse-client-r/issues/125)
It ended on February 1st 

The thing that correlates to the recovery was the start of the new month.
We don't understand what exactly caused the issue which means it could happen again.

There is information on the incident on [this thread](https://iqss.slack.com/archives/C03R1E7T4KA/p1674840525617389) in Slack

**Comment by mreekie** [comment]
Just an assertion:
- I don't think this was fixed by the CC update.
- The CC was updated after the Actions had already begun to work again.
- The cc had been expired since 2018.

**Comment by pdurbin** [comment]
> How can you tell how many hours of GitHub Actions you used last month? It's non-obvious.

This might help? https://fergusmacdermot.medium.com/github-action-usage-minute-challenges-and-an-action-that-solves-e0224478857c

Hopefully there's an easier way. 😅

**Comment by poikilotherm** [comment]
Actually this is really easy to do and looks faboulus! It even gives an overview per workflow!

https://github.com/marketplace/actions/github-actions-usage-audit

Let me know if you want a helping hand.

**Comment by pdurbin** [comment]
@poikilotherm sure please make a PR for a GitHub Action to track how many GitHub Action minutes we have left. 😄 

Meanwhile, I'm noticing "GitHub Actions is not available for private repositories on legacy per-repository billing plans. Review your pricing plan to get access to this feature" under IQSS and an nice report under gdcc.

## IQSS (no report, message about legacy billing plan)

<img width="1386" alt="Screen Shot 2023-02-02 at 7 06 54 AM" src="https://user-images.githubusercontent.com/21006/216320973-949ca5dd-0429-4e52-80cc-f58dc09d76b1.png">

## gdcc (nice report of GitHub Actions usage)

<img width="1386" alt="Screen Shot 2023-02-02 at 7 06 27 AM" src="https://user-images.githubusercontent.com/21006/216320998-92a0cf1a-6aaf-4bda-818a-3840e0379084.png">

**Comment by poikilotherm** [comment]
Maybe this is important:

![grafik](https://user-images.githubusercontent.com/5468333/216322315-09c51d01-6dbe-4acc-8855-8eefc95330be.png)

From https://github.com/organizations/gdcc/billing/plans

**Comment by pdurbin** [comment]
@poikilotherm yes, my understanding is that for public, you get 2000 minutes per month free.

> This might help? https://fergusmacdermot.medium.com/github-action-usage-minute-challenges-and-an-action-that-solves-e0224478857c

More on this tool:

## What is the Reason for this Action?

This action came about because a project I was working on kept running out of free minutes very quickly. As an admin I
was asking teams to reduce usage, but they had no idea which repos and workflows were clocking up the minutes. Worse,
the billing user was away for a day, the free minutes ran out, and we had no buffer set up. Development and deployment
came to a virtual standstill. So I started investigating ways to make usage more transparent.

With the use of GitHub Actions (GHA) in build and deploy pipelines, usage adda up very quickly. Once the monthly limit
has been reached, as we found out, the workflows will just stop running thereby causing huge disruption in the build and
delivery pipeline. If the billing user is on holiday, this is pretty disastrous. Better to be forewarned and so extra
credits can be added by the billing owner.

However, the usage minutes total is hidden away in the admin section and so repo owners cannot even see GHA usage. Even
if they could, the usage CSV that GitHub sends out contains too much information making it hard to isolate heavy usage
workflows and repos.

Furthermore, MacOS usage is charged at 10x, Windows 2x the rate of Ubuntu machines. This means that a 7 second action
runtime will be consume 1 minute of the allowance on Ubuntu, 10 minutes on MacOS, and 2 minutes on Windows. MacOS builds
can take 20-30 minutes and the free minutes soon dry up. It turned out, this was the cause of our high usage. Top tip:
don't build MacOS machines in GitHub.

From https://github.com/fergusmacd/github-actions-usage or https://github.com/marketplace/actions/github-actions-usage-audit

**Comment by wibeasley** [comment]
> MacOS usage is charged at 10x, Windows 2x the rate of Ubuntu machines

No way --thanks for finding that info.  I have never seen it.  I was wondering if the nonfree OSes cost GitHub anything.  This makes sense.

**Comment by pdurbin** [comment]
@wibeasley yeah, pretty interesting.

> This might help? https://fergusmacdermot.medium.com/github-action-usage-minute-challenges-and-an-action-that-solves-e0224478857c

I talked to @poikilotherm about this and he recommends against putting the check in the main "dataverse" repo:

"Not sure that's a good idea. Everyone syncing from develop etc would get it. Also we could make sure to only run it in our repo, just like we did for base container"

I'm not sure what repo to put it in though. A new repo? It certainly doesn't need to have "dataverse" in the name because the limits affect all repos under IQSS that use GitHub Actions, including https://github.com/IQSS/clarify and https://github.com/IQSS/Amelia.

As I said to @mreekie we're flying blind. We have no insight into how many hours we're using. Someone might need to contact GitHub support. Maybe flying blind is tied in with our legacy plan? (See the screenshot above.) I'm really not sure.
